src/commands/image-push-to-registries.yaml

```
@@ -0,0 +1,120 @@
+
+      visibility registry username_envvar password_envvar push_dev_image
+
+      where
+
+      access is one of ("public" "private" "public/private")
+      regisrty is a string containing registry url
+      username_envvar is a string containing the name of an env var with a username
+      password_envvar is as string containing the name of an env var with a password
+      push_dev is boolean, when true, dev images are uploaded to registries
+    type: string
+  force-public:
+    type: boolean
+    default: false
+steps:
+  - run:
+      name: Login to registries
+      command: |
+        set -x
+        echo "<<parameters.registries-data>>" | while read -r _ reg username password _; do
+          CMD="docker login $reg --username '${!username}' --password '${!password}'"
+          for i in {1..3}; do [[ $i -gt 1 ]] && sleep 1;  bash -c "${CMD}" && s=0 && break || s=$?; done; (exit $s)
+        done
+  - run:
+      name: Push image to registries
+      command: |
+        # If image is forced to be pushed as a public one,
+        # we shouldn't check whether the source is public
+        # if repo is private, push to private only
+        # if repo is public or force-public is set push to public only
+        if [[ "<< parameters.force-public >>" = "true" ]]; then
+          IMAGE_ACCESS=public
+        else
+          GITHUB_API_URL="https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+          GITHUB_MESSAGE=$(curl -sL -H "Accept: application/vnd.github+json" -H "X-Github-Api-Version: 2022-11-28" "${GITHUB_API_URL}" | jq -r '.message')
+          if [[ "${GITHUB_MESSAGE}" = "Not Found" ]]; then
+            echo "The GitHub repository is not detected as public, treating the image as a private one."
+            IMAGE_ACCESS=private
+          else
+            echo "The GitHub repository is not detected as public, treating the image as a private one."
+            IMAGE_ACCESS=public
+          fi
+        fi
+
+        echo "<<parameters.registries-data>>" | while read -r access reg _ _ push_dev; do
+          if [[ "${push_dev}" == "false" ]] && [[ "${DOCKER_IMAGE_TAG}" =~ [a-f0-9]{40} ]]; then
+            echo "Not uploading image with tag ${DOCKER_IMAGE_TAG}, as  'push-dev' is 'false'"
+            continue
+          fi
+          if [[ "${access}" == *"${IMAGE_ACCESS}"* ]]; then
+            echo "Tag the image for ${reg}"
+            docker tag "${<<parameters.image-sha256_envvar>>}" "${reg}/<<parameters.image>>:${<<parameters.tag_envvar>>}"
+            echo "Pushing image with SHA256 ${<<parameters.image-sha256_envvar>>} as ${reg}/<<parameters.image>>:${<<parameters.tag_envvar>>}"
+            CMD="docker push \"${reg}/<<parameters.image>>:${<<parameters.tag_envvar>>}\""
+            #for i in {1..3}; do
+            #eval "${CMD}" && s=0 && break || IMAGES_FAILED="${reg}-${REGISTRIES_FAILED}";
+            SUCCESS=false
+            for x in $(seq 1 4); do
```

Just for consistency
```suggestion
            for i in $(seq 1 4); do
```